### PR TITLE
[closes #34] Implement basic test for oauth settings page

### LINKED ISSUE #34: Implement test for Global Settings
The preview of the page is available in latest builds and also present on `syndesis-qe` instance. It's a good starting point to begin with e2e test impl.

### PR DESCRIPTION
**Pull request by jludvice**
@mmelko @lvydra @dsimansk 

This is very basic test for settings page.
I didn't do deleting of settings as it throws HTTP500 (reported as syndesis-rest/issues#549)
and it wasn't clear how it was supposed to work.
Delete settings is TODO for future

### CODE REVIEW

File: e2e/settings/settings.steps.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  @when(/^"([^"]*)" clicks to the "([^"]*)" item "([^"]*)" button$/)
+  public clickSettingsButton(userAlias: string, itemTitle: string, buttonTitle: string): P<any> {
+    const page = new SettingsPage();
+    return page.clickButton(itemTitle, buttonTitle);
+  }
+
+  @when(/^fill form in "([^"]*)" settings item$/)
+  public fillSettingsItemForm(itemTitle: string): P<any> {
+    const settings = new OAuthSettingsComponent();
+    const toFill = this.world.testConfig.settings[itemTitle];
```
Comment (jludvice) [inline e2e/settings/settings.steps.ts]: This test requires update to test settings config map.
I already did that on syndesis-ci.

File: e2e/settings/settings.po.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    return item.element(by.buttonText(buttonTitle)).click();
+  }
+}
+
+
+/**
+ * Represents oauth apps tab.
+ */
+export class OAuthSettingsComponent extends SettingsPage {
+  rootElement(): ElementFinder {
```
Comment (jludvice) [inline e2e/settings/settings.po.ts]: I'm not sure inheritance is good approach here.
What do you think about it?

File: e2e/settings/settings.po.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+  /**
+   * Get title of active settings tab
+   * @returns {Promise<string>} title of active settings tab
+   */
+  async activeTab(): P<string> {
+    const activeTab = this.rootElement().$('ul.nav-tabs li.active a');
+    return activeTab.getText();
+  }
```
Comment (jludvice) [inline e2e/settings/settings.po.ts]: I hope that other tabs will have similar structure and methods `getSettingsItem` and `clickButton` will work also for them.

File: e2e/settings/settings.po.ts (hunk 2 above)
Comment (lvydra) [reply e2e/settings/settings.po.ts]: I'm not sure to. Oauth apps tab as part of setting page, defined by 'syndesis-settings-root' element, could be better as SettingPage attribute.  But still we could use some kind of general ancestor for tabs.

Comment (jludvice) [reply e2e/settings/settings.po.ts]: @lvydra can we leave it this way for some time?
I bet we'll have to rewrite it anyway once engineering implements another Tab on this page.

**Pull request merged**